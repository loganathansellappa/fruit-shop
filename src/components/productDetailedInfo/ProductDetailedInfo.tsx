import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Grid,
  LinearProgress,
  TextField,
} from '@material-ui/core';
import { Title, Wrapper } from './ProductDetailedInfo.styles';
import React, { useRef, useState } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import {
  useEditProductMutation,
  useGetProductsQuery,
} from '../../services/Product';
import Error from '../error/Error';
import { ProductDetail } from '~/@types/Product';

const ProductDetailedInfo = () => {
  const { id = 1 } = useParams();
  const navigate = useNavigate();
  const {
    data,
    error,
    isLoading,
    isFetching: isFetching,
  } = useGetProductsQuery(`/products/${id}`);
  const titleText = useRef();

  const product = data as ProductDetail;
  const [open, setOpen] = React.useState(false);
  const [validationError, setValidationError] = useState<string>('');

  const [updateTitle] = useEditProductMutation();

  const handleUpdate = async () => {
    const { value: title } = titleText.current;
    if (title.length) {
      const data = {
        id: product.id,
        title,
      };
      const { error } = await updateTitle(data);
      if (error?.data?.message?.length) {
        setValidationError(error?.data?.message);
      } else {
        setOpen(false);
        setValidationError('');
      }
    } else {
      setValidationError('Cant be Blank');
    }
  };

  const handleClose = () => {
    setValidationError('');
    setOpen(false);
  };

  if (error) {
    return <Error owner={true} />;
  }
  if (isFetching || isLoading) {
    return <LinearProgress style={{ margin: 100 }} />;
  }
  return (
    <Wrapper>
      <img src={product.imageUrl} alt={product.title} />
      <Title onClick={() => setOpen(true)}>
        <h3>Title</h3>
        <p>{product.title}</p>
      </Title>
      <div>
        <h3>price</h3>
        <p>â‚¬{product.price}</p>
      </div>

      <Grid container justifyContent={'flex-start'}>
        <h3>description</h3>
        <p>{product.description}</p>
      </Grid>
      <div>
        <h3>Discounts {product.discount?.id ? 'Available' : 'Unavailable'} </h3>
      </div>
      <Grid container alignItems={'center'} justifyContent={'center'}>
        <Grid item xs={2}>
          <Button variant="outlined" onClick={() => navigate('/admin')}>
            Back
          </Button>
        </Grid>
      </Grid>
      <Dialog
        disableEnforceFocus
        fullWidth
        open={open}
        onClose={() => handleClose()}
      >
        <DialogTitle>Update Product Title</DialogTitle>
        <DialogContent>
          <DialogContentText>{product.type}</DialogContentText>
          <TextField
            inputRef={titleText}
            error={validationError.length > 0}
            helperText={validationError}
            autoFocus
            margin="dense"
            id="name"
            label="Title"
            fullWidth
            variant="standard"
          />
        </DialogContent>
        <DialogActions>
          <Button variant="outlined" onClick={handleClose}>
            Cancel
          </Button>
          <Button variant="outlined" onClick={handleUpdate}>
            Update
          </Button>
        </DialogActions>
      </Dialog>
    </Wrapper>
  );
};

export default ProductDetailedInfo;
