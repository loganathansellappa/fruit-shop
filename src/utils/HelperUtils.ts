import { Discount, ProductDetail } from '~/@types/Product';

export const apiService = () => `${process.env.SHOP_URL}`;
export const truncateString = (string = '', maxLength = 20) =>
  string.length > maxLength ? string.split(' ').slice(0, 20).join(' ') : string;
export const getDiscount = (product: ProductDetail, discounts: Discount[]) =>
  discounts.find((d) => product.type === d.product_type);
