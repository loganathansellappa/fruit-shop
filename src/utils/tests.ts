import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';

export const renderWithRouter = (ui: JSX.Element, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

export const mockResource = () => {
  return {
    discounts: [
      {
        id: 1,
        valid: true,
        type: 'Quantity',
        product_type: 'Apple',
        value: 2,
        reward: 1,
      },
    ],
    prodWithDiscount: {
      id: 2,
      type: 'Apple',
      imageUrl:
        'https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg',
      title: 'Apple - 2',
      description:
        'AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
      price: 1,
    },
    prodWithoutDiscount: {
      id: 2,
      type: 'Banana',
      imageUrl:
        'https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg',
      title: 'Banana',
      description:
        'AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
      price: 2,
    },
    listResponse: {
      products: [
        {
          id: 1,
          type: 'Banana',
          imageUrl:
            'https://avectime.com/grocery&gourmet/agriculture/brands/images/gros_michel_banana/gros_michel_banana1.jpg',
          title: 'Banana - 1',
          description:
            'BananaLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 2,
        },
        {
          id: 2,
          type: 'Apple',
          imageUrl:
            'https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg',
          title: 'Apple - 2',
          description:
            'AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 1,
        },
        {
          id: 3,
          type: 'Pineapple',
          imageUrl:
            'https://helios-i.mashable.com/imagery/articles/05W5DssM7oLPbBjiU4ZY6ob/hero-image.fill.size_1248x702.v1645798494.jpg',
          title: 'Pineapple - 3',
          description:
            'PineappleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 5,
        },
        {
          id: 4,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 4',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 25,
        },
        {
          id: 5,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 5',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 2,
        },
        {
          id: 6,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 6',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 29,
        },
        {
          id: 7,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 7',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 15,
        },
        {
          id: 8,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 8',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 9,
        },
        {
          id: 9,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 9',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 17,
        },
        {
          id: 10,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 10',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 21,
        },
      ],
      discounts: [
        {
          id: 53,
          valid: true,
          type: 'Quantity',
          product_type: 'Apple',
          value: 2,
          reward: 1,
        },
        {
          id: 54,
          valid: true,
          type: 'Price',
          product_type: 'Pineapple',
          value: 2,
          reward: 3,
        },
      ],
      total: 52,
    },
    cart: {
      products: {
        Banana: {
          id: 1,
          type: 'Banana',
          imageUrl:
            'https://avectime.com/grocery&gourmet/agriculture/brands/images/gros_michel_banana/gros_michel_banana1.jpg',
          title: 'Banana - 1',
          description:
            'BananaLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 2,
          totalQuantity: 4,
          totalPrice: 8,
          discountPrice: 0,
          discountQuantity: 0,
          finalQuantity: 4,
          finalPrice: 8,
        },
        Apple: {
          id: 2,
          type: 'Apple',
          imageUrl:
            'https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg',
          title: 'Apple - 2',
          description:
            'AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 1,
          totalQuantity: 4,
          totalPrice: 4,
          discountPrice: 0,
          discountQuantity: 2,
          finalQuantity: 6,
          finalPrice: 4,
        },
      },
      totalQuantity: 10,
      totalPrice: 12,
      totalUserQuantity: 8,
    },
  };
};
