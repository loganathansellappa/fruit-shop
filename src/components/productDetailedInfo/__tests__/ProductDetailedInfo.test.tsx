import { mockResource, renderWithRouter } from '../../..//utils/tests';
import ProductDetailedInfo from '../ProductDetailedInfo';
import { waitFor } from '@testing-library/react';

const mockResponse = {
  data: { ...mockResource().prodWithDiscount },
};

jest.mock('../../../services/Product', () => ({
  __esModule: true,
  useGetProductsQuery: () => mockResponse,
  useEditProductMutation: () => {
    const td = () =>
      Promise.resolve({
        data: {
          ...mockResource().prodWithDiscount,
          discount: mockResource().discounts[0],
        },
      });
    return [td];
  },
}));

describe('ProductDetailedInfo', () => {
  let element: { container: HTMLElement };

  const match = { params: { id: mockResource().prodWithoutDiscount.id } };

  beforeEach(() => {
    element = renderWithRouter(<ProductDetailedInfo match={match} />);
  });

  test('renders ProductDetailedInfo', async () => {
    await waitFor(() => {
      expect(element.container.firstChild).toMatchSnapshot();
      expect(element.container.textContent).toContain(
        mockResource().prodWithDiscount.title
      );
    });
  });
});
