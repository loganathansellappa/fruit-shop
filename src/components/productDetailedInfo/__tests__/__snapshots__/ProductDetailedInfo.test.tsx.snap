// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailedInfo renders ProductDetailedInfo 1`] = `
<div
  class="sc-bczRLJ inpPbK"
>
  <img
    alt="Apple - 2"
    src="https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg"
  />
  <div
    class="sc-gsnTZi cMbbcs"
  >
    <h3>
      Title
    </h3>
    <p>
      Apple - 2
    </p>
  </div>
  <div>
    <h3>
      price
    </h3>
    <p>
      â‚¬
      1
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <h3>
      description
    </h3>
    <p>
      AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.
    </p>
  </div>
  <div>
    <h3>
      Discounts 
      Unavailable
       
    </h3>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
        href="/admin"
      >
        Back
      </a>
    </div>
  </div>
</div>
`;
